javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/instrumentation/indy/InstrumentationModuleClassLoader.java

```diff
@@ -0,0 +1,252 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.tooling.instrumentation.indy;
+
+import java.io.IOException;
+import java.lang.invoke.MethodHandles;
+import java.lang.reflect.Method;
+import java.net.URL;
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+import java.security.ProtectionDomain;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/**
+ * Classloader used to load the helper classes from {@link
+ * io.opentelemetry.javaagent.extension.instrumentation.InstrumentationModule}s, so that those
+ * classes have access to both the agent/extension classes and the instrumented application classes.
+ *
+ * <p>This classloader implements the following classloading delegation strategy:
+ *
+ * <ul>
+ *   <li>First, injected classes are considered (usually the helper classes from the
+ *       InstrumentationModule)
+ *   <li>Next, the classloader looks in the agent or extension classloader, depending on the where
+ *       the InstrumentationModule comes from
+ *   <li>Finally, the instrumented application classloader is checked for the class
+ * </ul>
+ *
+ * <p>In addition, this classloader ensures that the lookup of corresponding .class resources follow
+ * the same delegation strategy, so that bytecode inspection tools work correctly.
+ */
+public class InstrumentationModuleClassLoader extends ClassLoader {
+
+  private static final Map<String, ClassCopySource> ALWAYS_INJECTED_CLASSES;
+
+  static {
+    Map<String, ClassCopySource> alwaysInjected = new HashMap<>();
+    alwaysInjected.put(
+        LookupExposer.class.getName(), ClassCopySource.create(LookupExposer.class).cached());
+    ALWAYS_INJECTED_CLASSES = Collections.unmodifiableMap(alwaysInjected);
+  }
```

Comment:
Since there's just a single key/value this can be initialized like so:
```suggestion
  private static final Map<String, ClassCopySource> ALWAYS_INJECTED_CLASSES =
      Collections.unmodifiableMap(
          Collections.singletonMap(
              LookupExposer.class.getName(), ClassCopySource.create(LookupExposer.class).cached()));
```